packages/block-library/src/navigation/edit/navigation-menu-selector.js

```diff
@@ -96,26 +79,40 @@ function NavigationMenuSelector( {
 	const menuUnavailable =
 		hasResolvedNavigationMenus && currentMenuId === null;
 
-	useEffect( () => {
-		if ( ! hasResolvedNavigationMenus ) {
-			setSelectorLabel( __( 'Loading …' ) );
-		} else if ( noMenuSelected || noBlockMenus || menuUnavailable ) {
-			setSelectorLabel( __( 'Select menu' ) );
-			setEnableOptions( shouldEnableMenuSelector );
-		}
+	let selectorLabel = '';
 
+	if ( isCreatingMenu || isResolvingNavigationMenus ) {
+		selectorLabel = __( 'Loading …' );
+	} else if ( noMenuSelected || noBlockMenus || menuUnavailable ) {
+		selectorLabel = __( 'Select menu' );
+	} else {
+		// Current Menu's title.
+		selectorLabel = currentTitle;
+	}
+
+	const shouldEnableMenuSelector =
+		( canSwitchNavigationMenu || canUserCreateNavigationMenu ) &&
+		hasResolvedNavigationMenus;
+
+	let enableOptions = false;
+
+	if ( isCreatingMenu ) {
+		enableOptions = false;
+	} else if ( shouldEnableMenuSelector ) {
+		enableOptions = true;
+	}
```

Comment:
```suggestion
	} else {
		enableOptions = shouldEnableMenuSelector;
	}
```

We could go for this route.